abcclassroom/tests/test_clone.py

```diff
@@ -32,6 +35,37 @@ def test_files(default_config, tmp_path):
             Path(assignment_path, f).touch()
 
 
+# TODO: Test that the correct message is returned when no course_materials dir
+# exists in the config
+
+
+def test_roster_missing(sample_course_structure, tmp_path):
+    """Test that when the roster is missing things fail gracefully."""
+    # config = sample_course_structure
+    assignment_name = "test_assignment"
+
+    with pytest.raises(FileNotFoundError, match="Cannot find roster file"):
+        abcclone.clone_repos(assignment_name, skip_existing=False)
+
+
+# TODO: Test what happens when the roster exists but has the incorrect format
+def test_roster_wrong_format(sample_course_structure, tmp_path):
+    """Test that when the roster is missing things fail gracefully."""
+
```

Comment:
This is bizarre because i swear i'm making a new blank roster - how the heck is this running ??

Reply:
i've tested it and it's like when this runs in a loop, it doesn't catch the keyerror... so it's simply returning nothing and failing quietly. i'm not sure what else to try here but am open to some feedback @kcranston  perhaps we can chat in our meeting today?

Reply:
resolving! fixed this